hack/verify-flags-underscore.py

```diff
@@ -95,8 +95,23 @@ def normalize_files(rootdir, files):
 def line_has_bad_flag(line, flagre):
     results  = flagre.findall(line)
     for result in results:
-        if "_" in result:
-            return True
+        if not "_" in result:
+            return False
+        # this should exclude many cases where jinja2 templates use kube flags
+        # as variables, except it uses _ for the variable name
+        if "{% set" + result + "= \"" in line:
+            return False
+        if "pillar[" + result + "]" in line:
+            return False
+        if "grains" + result in line:
+            return False
+        # These are usually yaml definitions
+        if result.endswith(":"):
+            return False
+         # something common in juju variables...
+        if "template_data[" + result + "]" in line:
+            return False
```

Comment:
Not by me...   I ran out of things that could cut down the whitelist significantly.  But anyone who looks at the exceptions list might find more ideas....